Tests/OptimizelyTests-Batch-iOS/EventDispatcherTests_Batch.swift

```diff
@@ -233,8 +230,8 @@ extension EventDispatcherTests_Batch {
     }
 
     func testBatchingEventsWhenProjectIdsNotEqual() {
-        // projectId change will flush all pending events, so this scenario should not happen frequently.
-        // but still possible when previous flushes failed.
+        // Batching set a boundary when an event has a different projectId from previous events.
+        // The remaining events will be sent as a separate batch (see "testFlushEventsWhenBatchFails()" for that).
```

Comment:
Does this mean that when new events arrive, those will be sent before the previous events? is this acceptable?

Reply:
New events will be appended to the end of the batch queue. They'll be batched (with a projectId and revision boundaries) in FIFO order, so the dispatch order is preserved.